src/main/java/org/broadinstitute/hellbender/engine/ReadShard.java

```diff
@@ -0,0 +1,182 @@
+     */
+    public SimpleInterval getInterval() {
+        return interval;
+    }
+
+    /**
+     * @return the interval this shard spans, potentially with additional padding on each side
+     */
+    public SimpleInterval getPaddedInterval() {
+        return paddedInterval;
+    }
+
+    /**
+     * @return an iterator over reads in this shard, as filtered using the configured read filter;
+     *         reads are lazily loaded rather than pre-loaded
+     */
+    @Override
+    public Iterator<GATKRead> iterator() {
+        final Iterator<GATKRead> readsIterator = readsSource.query(paddedInterval);
+
+        return readFilter != null ? new ReadFilteringIterator(readsIterator, readFilter) : readsIterator;
+    }
+
+    /**
+     * @return a List containing all reads in this shard, pre-loaded, and filtered using the configured read filter
+     *
+     * note: call {@link #iterator} instead to avoid pre-loading all reads at once
+     */
+    public List<GATKRead> loadAllReads() {
+        return StreamSupport.stream(Spliterators.spliteratorUnknownSize(iterator(), Spliterator.ORDERED), false).collect(Collectors.toList());
+    }
+
+    /**
+     * @param loc Locatable to test
+     * @return true if loc is contained within this shard's interval, otherwise false
+     */
+    public boolean isContainedWithinShard( final Locatable loc ) {
+        return interval.contains(loc);
+    }
+
+    /**
+     * @param loc Locatable to test
+     * @return true if loc starts within this shard's interval, otherwise false
+     */
+    public boolean startsWithinShard( final Locatable loc ) {
+        return interval.contains(new SimpleInterval(loc.getContig(), loc.getStart(), loc.getStart()));
+    }
+
+    /**
+     * @return number of bases of padding to the left of our interval
+     */
+    public int numLeftPaddingBases() {
+        return interval.getStart() - paddedInterval.getStart();
+    }
+
+    /**
+     * @return number of bases of padding to the right of our interval
+     */
+    public int numRightPaddingBases() {
+        return paddedInterval.getEnd() - interval.getEnd();
+    }
+
+    /**
+     * Shard an interval into ReadShards using the shardSize and shardPadding arguments
+     *
+     * @param interval interval to shard; must be on the contig according to the provided dictionary
+     * @param shardSize desired shard size; intervals larger than this will be divided into shards of up to this size
+     * @param shardPadding desired shard padding; each shard's interval will be padded on both sides by this number of bases
+     * @param readsSource data source for reads
+     * @param dictionary sequence dictionary for reads
+     * @return List of {@link ReadShard} objects, sharded and padded as necessary
+     */
+    public static List<ReadShard> shardInterval( final SimpleInterval interval, final int shardSize, final int shardPadding, final ReadsDataSource readsSource, final SAMSequenceDictionary dictionary ) {
```

Comment:
Even if the `SlidingWindowWalker` is not going to be used for the `AssemblyRegionWalker`, I think that the `ReadShard` could be useful in terms of performance for it. In that case, the method will be needed. Padding could be performed in sliding-windows too.

My idea wasn't remove this method, it was mos likely to do the following with it:

```java
public static List<ReadShard> shardInterval( final SimpleInterval interval, final int shardSize, final int shardPadding, final ReadsDataSource readsSource, final SAMSequenceDictionary dictionary ) {
    return shardInterval(interval, shardSize, shardSize, shardPadding, readsSource, dictionary);
}

public static List<ReadShard> shardInterval( final SimpleInterval interval, final int shardSize, final int shardStep, final int shardPadding, final ReadsDataSource readsSource, final SAMSequenceDictionary dictionary ) {
    // unchanged code until line 1777
    start += shardStep;
    // unchanged code untill the end
}
````

This will provide the same behaviour for the rest of the commit, but allows other way of generate the shard intervals. Of course, it could be done in other part of the code. But of course, I won't use `ReadShard` in `SlidingWindowWalker` if it is goin to be hidden, because it will be one of the parameters of the `apply` function.

Just let me know what is the plan for `ReadShard` for implement #1708 in a different way. And many thanks for the help and fruitful discussion about this topic.